Xamarin.Essentials/Platform/Platform.ios.cs

```diff
@@ -57,5 +57,8 @@ internal static UIViewController GetCurrentViewController(bool throwIfNull = tru
 
         internal static CMMotionManager MotionManager =>
             motionManager ?? (motionManager = new CMMotionManager());
+
+        internal static NSOperationQueue CurrentQueue =>
+            NSOperationQueue.CurrentQueue ?? new NSOperationQueue();
```

Comment:
We just have to remember (or maybe make a note) that this is going to fail:

```csharp
var myThing = DoSomething(Platform.CurrentQueue);

if (myThing.Queue == Platform.CurrentQueue) {
    // in my mind
} else {
    // say whaa?!?
}
```

Reply:
Maybe we should change this from a property to a method like `Platform.GetQueue()` or something that indicates that it does not return the same object in some cases.

Reply:
But, this is internal so a comment may be just fine as well.

Reply:
yeah, i like get queue so it doesn't think it is always the same. Good call.